src/Polymesh.ts

```diff
@@ -76,37 +104,52 @@ export class Polymesh {
       });
 
       harvesterClient = new ApolloClient({
```

Comment:
I think it makes more sense to have `harvesterClient` here defined as `ApolloClient<NormalizedCacheObject> | null` and initialized as null. You then only instance the apollo client and re-assign it to that variable IF the harvester config was passed, and instead of having the `isApolloConfigured` variable, the `Context.create` function should just deduce that from the harvesterClient being null or not

Reply:
Behind the scenes in the Context class, you save the harvester client in a private _harvesterClient variable, and define

```
public get harvesterClient()
```

Which throws an error if the variable isn't defined and returns it if it is.

To the user it's totally transparent and we don't even need to use the Proxy pattern